openedx/frontend-app-authoring
# Open Editors in a Modal (library components only) [FC-0062]

## Description

Implements https://github.com/openedx/frontend-app-authoring/issues/1321

This PR changes the built-in editors so that in libraries they open in a modal, with the library partially visible behind them. Because we still have to support the legacy unit page as default, editing content in courses will continue to use the full screen editor.

Screenshots:

| Editing a library component | Editing a course component (no change) |
|--|--|
| ![Screenshot 1](https://github.com/user-attachments/assets/93152c1b-81f4-47d6-ab1d-3eaff475f29e) |  ![Screenshot 2](https://github.com/user-attachments/assets/564cb81e-2146-4796-9700-f42e4a1f2df8) |

## Testing instructions

Try the various (Text, Problem, Video) editors in libraries as well as in courses, and make sure there are no regressions.

## Other information

Includes changes from https://github.com/openedx/frontend-app-authoring/pull/1345 - merge that one first.

Private ref: [FAL-3861](https://tasks.opencraft.com/browse/FAL-3861).
